client/sidebar/header/CreateTeam.js

```diff
@@ -0,0 +1,224 @@
+import { FlowRouter } from 'meteor/kadira:flow-router';
+import React, { memo, useCallback, useEffect, useMemo, useState } from 'react';
+import { useMutableCallback, useDebouncedCallback } from '@rocket.chat/fuselage-hooks';
+import { Box, Modal, ButtonGroup, Button, TextInput, Icon, Field, ToggleSwitch } from '@rocket.chat/fuselage';
+
+import { useTranslation } from '../../contexts/TranslationContext';
+import { useForm } from '../../hooks/useForm';
+import { useEndpointActionExperimental } from '../../hooks/useEndpointAction';
+import UserAutoCompleteMultiple from '../../../ee/client/audit/UserAutoCompleteMultiple';
+import { useSetting } from '../../contexts/SettingsContext';
+import { usePermission } from '../../contexts/AuthorizationContext';
+import { useMethod } from '../../contexts/ServerContext';
+
+export const CreateTeam = ({
+	values,
+	handlers,
+	hasUnsavedChanges,
+	onChangeMembers,
+	onChangeType,
+	onChangeBroadcast,
+	canCreateTeam,
+	e2eEnabledForPrivateByDefault,
+	onCreate,
+	onClose,
+}) => {
+	const t = useTranslation();
+	const e2eEnabled = useSetting('E2E_Enable');
+	const namesValidation = useSetting('UTF8_Names_Validation');
+	const allowSpecialNames = useSetting('UI_Allow_room_names_with_special_chars');
+	const teamNameExists = useMethod('roomNameExists');
+	const teamNameRegex = useMemo(() => {
+		if (allowSpecialNames) {
+			return '';
+		}
+		const regex = new RegExp(`^${ namesValidation }$`);
+
+		return regex;
+	}, [allowSpecialNames, namesValidation]);
+
+	const [nameError, setNameError] = useState();
+
+	const checkName = useDebouncedCallback(async (name) => {
+		setNameError(false);
```

Comment:
Use a single type! `string` or `boolean`, choose one and stick with it.

Reply:
Ok!